### PR TITLE
fixed iseq runs to work with the miniseq parser

### LINKED ISSUE #53: Add iseq parser
## Describe your idea for a new feature
iseq is almost identical to miniseq but does not include a `Description` field, this causes issue in some cases when trying to use the miniseq parser to upload iseq runs.

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## Description of changes
Made `iseq` runs to be parsable by the `miniseq` parser
Added `iseq` as a keyword to select the `miniseq` parser from command line and gui
Changed wording in docs to show that iseq is now supported
Added unittest to test for edge case where the miniseq parser would previously fail when attempting to parse an iseq run

## Related issue
Fixes #53 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: tests/parsers/miniseq/test_sample_parser.py [hunk 1]
```diff
@@ -492,3 +492,39 @@ def test_parse_out_sequence_file(self):
...
+        # Includes a single project
+        self.assertEqual(len(sequencing_run.project_list), 1)
+        # is of type Project
+        self.assertEqual(type(sequencing_run.project_list[0]), model.Project)
+        # Project has 3 samples
+        self.assertEqual(len(sequencing_run.project_list[0].sample_list), 3)
+        # samples are of type Sample
+        self.assertEqual(type(sequencing_run.project_list[0].sample_list[0]), model.Sample)
+        # samples have correct description
+        self.assertEqual(sequencing_run.project_list[0].sample_list[0].description, "desc1")
```
Comment (tom114) [inline tests/parsers/miniseq/test_sample_parser.py]: Maybe I'm not understanding, but didn't you say iSeq doesn't have a description field?  Should we also be adding a test without a description field?

File: parsers/miniseq/sample_parser.py [hunk 2]
```diff
@@ -295,11 +295,13 @@ def _parse_samples(sample_sheet_file):
...
         new_sample_dict = deepcopy(sample_dict)
         new_sample_name = new_sample_dict['sampleName']
-        # Some versions of the illumina *seq software has description fields, and others do not
+        # Some versions of the illumina *seq software has Description fields, and others do not
         # If they do, we need to include the description field here (or else we end up with duplication of fields)
         # If they don't we give it a blank description
         new_sample_desc = ''
         if 'Description' in new_sample_dict:
+            new_sample_desc = new_sample_dict['Description']
+        if 'description' in new_sample_dict:
```
Comment (tom114) [inline parsers/miniseq/sample_parser.py]: What's effect does the difference between the capital `D` and small `d` description fields have?

File: tests/parsers/miniseq/test_sample_parser.py (hunk 1 above)
Comment (JeffreyThiessen) [reply tests/parsers/miniseq/test_sample_parser.py]: MiniSeq does not have a description field by default, but iSeq seems to sometimes have it and sometimes not. All the other tests in the MiniSeq sample_parser test suite are testing without description fields so I only added a test that is just like a regular MiniSeq run but with a description field.

File: parsers/miniseq/sample_parser.py (hunk 2 above)
Comment (JeffreyThiessen) [reply parsers/miniseq/sample_parser.py]: Depends how it comes off the sample sheet. I have seen both happen on the sequencers now, and I don't know if it's something weird with illumina software, or if it is operator error, but this covers the bases and makes sure that if something weird happens where two different description fields appear, it can continue.
